fs/vfs/main.cc

```diff
@@ -1245,6 +1245,39 @@ int link(const char *oldpath, const char *newpath)
     return -1;
 }
 
+OSV_LIBC_API
+int linkat(int olddirfd, const char *oldpath, int newdirfd, const char *newpath, int flags)
+{
+    if (flags & AT_SYMLINK_FOLLOW) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (!oldpath || !newpath) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (newpath[0] == '/' || newdirfd == AT_FDCWD) {
+        return vfs_fun_at2(olddirfd, oldpath, [newpath](const char *path) {
+            return link(path, newpath);
+        });
+    } else {
+        char absolute_newpath[PATH_MAX];
+        auto error = vfs_fun_at(newdirfd, newpath, [&absolute_newpath](const char *absolute_path) {
+            strcpy(absolute_newpath, absolute_path);
```

Comment:
Is this strcpy() safe? Do we know that absolute_path is shorter than PATH_MAX? If not, maybe should use strlcpy or something?

Reply:
Replaced with `strlcpy()` here and in `renameat()` where some code was stolen from.